tests/Unit/Evolution/Systems/Cce/Test_LinearSolve.cpp

```diff
@@ -0,0 +1,477 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details
+
+#include "tests/Unit/TestingFramework.hpp"
+
+#include <algorithm>
+
+#include "DataStructures/ComplexDataVector.hpp"
+#include "DataStructures/ComplexModalVector.hpp"
+#include "Evolution/Systems/Cce/LinearSolve.hpp"
+#include "Evolution/Systems/Cce/Tags.hpp"
+#include "Utilities/VectorAlgebra.hpp"
+#include "tests/Unit/Evolution/Systems/Cce/CceComputationTestHelpers.hpp"
+#include "tests/Unit/TestHelpers.hpp"
+#include "tests/Utilities/MakeWithRandomValues.hpp"
+
+namespace Cce {
+
+namespace {
+
+// the first several helper functions in this file are used to create and
+// process the polynomial data that will be used to create the expected values
+// in each of the integration routines
+
+ComplexDataVector radial_vector_from_power_series(
+    const ComplexModalVector& powers,
+    const ComplexDataVector& one_minus_y) noexcept {
+  ComplexDataVector result{one_minus_y.size(), powers[0]};
+  for (size_t i = 1; i < powers.size(); ++i) {
+    // use of TestHelpers::power due to an internal bug in blaze powers of
+    // Complex vectors
+    result += powers[i] * TestHelpers::power(one_minus_y, i);
+  }
+  return result;
+}
+
+template <typename BondiValueTag, typename DataBoxType>
+void make_boundary_data(const gsl::not_null<DataBoxType*> box,
```

Comment:
template on the `DbTagsList` and constrain the type to `db::DataBox<DbTagsList>`